streams/src/test/java/org/apache/kafka/streams/integration/RegexSourceIntegrationTest.java

```diff
@@ -322,6 +323,61 @@ public void testShouldReadFromRegexAndNamedTopics() throws Exception {
     }
 
     @Test
+    public void shouldAlwaysGetNewlyCreatedTopics() throws Exception {
+        final KStreamBuilder builderLeader = new KStreamBuilder();
+        final KStreamBuilder builderFollowerI = new KStreamBuilder();
+        final KStreamBuilder builderFollowerII = new KStreamBuilder();
+
+        final Pattern pattern = Pattern.compile("updating-\\d");
+
+        builderLeader.stream(pattern);
+        builderFollowerI.stream(pattern);
+        builderFollowerII.stream(pattern);
+
+        final KafkaStreams streamsLeader = new KafkaStreams(builderLeader, streamsConfiguration);
+        final KafkaStreams streamsFollowerI = new KafkaStreams(builderFollowerI,
+                                                               streamsConfiguration);
+        final KafkaStreams streamsFollowerII = new KafkaStreams(builderFollowerII,
+                                                                streamsConfiguration);
+
+        final int numberTopicsToAdd = 5;
+
+        try {
+            streamsLeader.start();
+            Thread.sleep(1000);
```

Comment:
I'm actually thinking more about making the unit test really "unit" by mocking stream partition assignor to always return unknown topic partitions; the current test case seems to be more "integration test"?